src/Metadata/Property/Factory/SerializerPropertyMetadataFactory.php

```diff
@@ -59,8 +59,12 @@ public function create(string $resourceClass, string $property, array $options =
 
         $propertyMetadata = $this->transformReadWrite($propertyMetadata, $resourceClass, $property, $normalizationGroups, $denormalizationGroups);
 
-        if (!$this->isResourceClass($resourceClass) && ($builtinType = $propertyMetadata->getBuiltinTypes()[0] ?? null) && $builtinType->isCollection()) {
-            return $propertyMetadata->withReadableLink(true)->withWritableLink(true);
+        if (!$this->isResourceClass($resourceClass) && ($types = $propertyMetadata->getBuiltinTypes())) {
```

Comment:
you could declare the `$types = $propertyMetadata->getBuiltinTypes()` outside of the `if` as it'll get used later if needed.